flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/controller/FlinkDeploymentController.java

```diff
@@ -86,26 +92,29 @@ public FlinkDeploymentController(
     @Override
     public DeleteControl cleanup(FlinkDeployment flinkApp, Context context) {
         LOG.info("Deleting FlinkDeployment");
+        OperatorUtils.updateStatusFromCache(flinkApp, statusCache);
```

Comment:
@gyfora This means to me we never give a chance to update the status outside the operator, correct?

Reply:
I think the user cannot update the status either. But in a way yes this means that only the Flink Operator can update the status that is correct.

Later when this is improved in the java operator sdk (version 3 ) we could rework this if we think it's important but this should not be a serious limiation.